packages/web/components/Setup/WizardPane.tsx

```diff
@@ -108,7 +116,41 @@ export const WizardPane = <T,>({
           });
         } else if (onSave) {
           setStatus('Saving…');
-          await onSave({ values, setStatus });
+          const images: HasuraImageSourcedProps = {};
+          // handle image upload of profileImageURL to web3.storage
+          if (values.profileImageURL) {
+            const formData = new FormData();
+
+            // 'profile' is the key for ceramic, equiveleant to profileImageURL in hasura
+            formData.append('profile', values.profileImageURL.file);
+            const result = await fetch(`/api/storage`, {
+              method: 'POST',
+              body: formData,
+              credentials: 'include',
+            });
+            const response = await result.json();
+            const { error } = response;
+            if (result.status >= 400 || error) {
+              throw new Error(
+                `web3.storage ${result.status} response: "${
+                  error ?? result.statusText
+                }"`,
+              );
+            }
+            images.profileImageURL = {
+              original: {
+                src: `ipfs://${response.profile}`,
+                mimeType: values.profileImageURL.file.type,
+                width: values.profileImageURL.width,
+                height: values.profileImageURL.height,
+              },
+            } as ImageSources;
+            // eslint-disable-next-line no-param-reassign
+            delete values.profileImageURL;
+            await onSave({ values, images, setStatus });
+          } else {
+            await onSave({ values, images, setStatus });
+          }
```

Comment:
In general if you're executing the same code at the end of both an `if` & `else`, I would think you could just execute it outside the conditional.

```suggestion
          }
          await onSave({ values, images, setStatus });
```